src/views/SourcesView.vue

```diff
@@ -1,250 +1,187 @@
-  isLoadingScreensAndWindows.value = true
-  await getVideoSources()
-  console.log(desktopCaptureSources.value)
-  await loadStreamSources()
-  isLoadingScreensAndWindows.value = false
-}
-
-/**
- * Get all the video sources available on the system
- */
-async function getVideoSources() {
-  // TODO: is not called again if new sources showes up
-  // Get the available video sources
-  desktopCaptureSources.value = await (window as any).electronAPI.getMedia()
+    // wait for the fetched screens and windows
+    desktopVideoStreamSources.value =
+        await streamHandler.getScreenAndApplicationSources();
+
+    // Loop through all sources and set the MediaStream to video nodes
+    Array.from(desktopVideoStreamSources.value).forEach(function (element) {
+        setSourceForVideoNode(element);
+    });
 }
 
-/**
- * Load the proper media stream source for the video element
- */
-async function loadStreamSources() {
-  // Is there a better way then this?
-  const videoElements: HTMLVideoElement | null =
-    document.querySelectorAll('.preview')
-
-  Array.from(videoElements).forEach(function (video, index) {
-    setSourceForVideoNode(desktopCaptureSources.value[index], video)
-  })
-}
-
-/**
- * Get the stream source
- * @param {string} sourceName
- */
-async function getStreamSource(source: any) {
-  const constraints: any = {
-    video: {
-      mandatory: {
-        chromeMediaSource: 'desktop',
-        chromeMediaSourceId: source.id,
-      },
-    },
-  }
-
-  // Create a Stream
-  const stream = await navigator.mediaDevices.getUserMedia(constraints)
-
-  return stream
-}
-
-/**
- * Load the stream sources
- * @param {any} source
- * @param {HTMLVideoElement} videoHTMLNode
- * @returns {Promise<void>}
- */
-async function setSourceForVideoNode(source: any, videoNode: HTMLVideoElement) {
-  const constraints: any = {
-    video: {
-      mandatory: {
-        chromeMediaSource: 'desktop',
-        chromeMediaSourceId: source.id,
-      },
-    },
-  }
-
-  // Create a Stream
-  const stream = await navigator.mediaDevices.getUserMedia(constraints)
-
-  // Preview the source in a video element
-  videoNode.srcObject = stream
-}
-
-/**
- * Load a specific stream source as main video
- *
- * @param {any} source
- */
-async function selectSource(source: any) {
-  // Set the selected source
-  selectedSource.value = source
-
-  // Preview the source in a video element
-  const videoElement: HTMLVideoElement | null =
-    document.querySelector('#mainVideo')
-
-  setSourceForVideoNode(source, videoElement)
+async function setSourceForVideoNode(source: any) {
+    const constraints: any = {
+        video: {
+            mandatory: {
+                chromeMediaSource: 'desktop',
+                chromeMediaSourceId: source.id,
+            },
+        },
+    };
+
+    streams.value.push(await navigator.mediaDevices.getUserMedia(constraints));
 }
 </script>
 
-<style lang="scss"></style>
+<style lang="scss" scoped></style>
```

Comment:
sources view still misses difference in structure between entire screens and single applications. I'd prefer a switch (see design: 
![grafik](https://user-images.githubusercontent.com/58397976/204328193-5c024499-9fa9-4641-9606-a439295b0951.png)

Reply:
I removed it because of some buggy things because the video wasn't showing.

Reply:
Can you try and re-add it before merge? This is some big difference to our design iteration and would have to be documented if missing.

Reply:
I can try but its kinda weird ngl 😂

Reply:
I‘m sure you will find a solution :)

Reply:
The solution is to do it later :)

Reply:
That ain't a solution...

Reply:
The issue #32 will work on this, we should focus on getting things done and get more things done.